h2o-py/tests/testdir_jira/pyunit_NOPASS_pubdev_6592_type_error.py

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+
+from tests import pyunit_utils
+
+# NOPASS, after fixing a bug it throws a correct error.
+# Note: the try-except interrupts another error behind this error, so it can´t be used in this test.
+def test_type_error():
+    from datatable import f
+    import h2o
+    h2o.init(0)
```

Comment:
Hmm, I don't think the test will work: 
after all, `h2o.init(0)` is supposed to throw an error, and what we are checking here is the behavior of the exception hook. Even catching the exception will not work, as we are interested in the behavior of the uncaught exception.

You could, however, launch this test in a separate python process:
```
try:
   subprocess.check_output(['python', '-c', 'from datatable import f; import h2o; h2o.init(0)'], stderr=subprocess.STDOUT)
except subprocess.CalledProcessError as e:
    text = e.output.decode()
    assert "Original exception was" not in text
    assert "cannot be cast to bool" not in text
    assert "Argument `url` should be a ?string, got integer 0" in text
    # etc...
```

Reply:
Also, I don't know whether `datatable` is a dependency of `h2o` already, but if it isn't then it probably doesn't make sense to introduce such a dependency for the sake of a single test...
Luckily, object `f` can easily be mimicked: just create a class with overloaded `__code__` attribute which would return an object with `__eq__` method that returns another object not castable to bool.